source/common/stream_info/utility.cc

```diff
@@ -1,29 +1,24 @@
 #include "common/stream_info/utility.h"
 
 #include <string>
+#include <type_traits>
 
 namespace Envoy {
 namespace StreamInfo {
 
-void ResponseFlagUtils::appendString(std::string& result, absl::string_view append) {
-  if (result.empty()) {
-    result = append;
-  } else {
-    result += ",";
-    result += append;
-  }
-}
-
 const std::string ResponseFlagUtils::toShortString(const StreamInfo& stream_info) {
-  std::string result;
-
+  absl::InlinedVector<absl::string_view,
+                      std::numeric_limits<std::underlying_type_t<ResponseFlag>>::digits>
```

Comment:
what is this number, btw? 

No idea what the impact of having (say) 100 here when you only need 3, but if that number is like 100000 you might blow out the stack.

Reply:
The underlying type of `enum class ResponseFlag` is int at my platform and the ::digit number is 31

Maybe some platform could use 64 or 63 but no more than that.
The stack should be safe IMO

Reply:
Can you add some comments? I don't immediately understand what this does. It seems like it would be more clear to just pick an arbitrary number like 1, since in most cases I think there will be a single flag.

Reply:
Yeah, usually 1 but InlinedVector is free. I can add the comment

Reply:
Sorry for being a readability pedant, but I just don't think using all of this templating stuff adds any value here? Can you just use 1, or 16 or whatever? This seems like a roundabout way to pick a small number.

/wait

Reply:
I like 4+. Then 16 since it's in your provided choices :)